tests/e2e/test_image.py

```diff
@@ -0,0 +1,499 @@
+                FROM ubuntu:latest
+                ADD {random_file_to_disable_layer_caching} /tmp
+                RUN echo !
+                """
+            )
+        )
+
+    tag = str(uuid.uuid4())
+
+    # WORKAROUND: Fixing 401 Not Authorized because of this problem:
+    # https://github.com/neuromation/platform-registry-api/issues/209
+    rnd = uuid.uuid4().hex[:6]
+    img_name = f"image:extras-e2e-custom-preset-{rnd}"
+    img_uri_str = f"{img_name}:{tag}"
+
+    try:
+        cmd = [
+            "neuro",
+            "image-build",
+            "--preset",
+            custom_preset,
+            "-e",
+            f"{dockerhub_auth_secret.name}=secret:{dockerhub_auth_secret.name}",
+            "-f",
+            str(dockerfile_path),
+            ".",
+            img_uri_str,
+        ]
+        result = cli_runner(cmd)
+        assert result.returncode == 0, result
+
+        result = repeat_until_success(["neuro", "image", "tags", img_name])
+        assert tag in result.stdout
+    finally:
+        try:
+            cli_runner(["neuro", "image", "rm", img_uri_str])
+        except Exception:
+            # Ignore exception
+            pass
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="kaniko does not work on Windows")
+def test_image_build_custom_dockerfile(
+    cli_runner: CLIRunner,
+    repeat_until_success: Callable[..., "CompletedProcess[str]"],
+    dockerhub_auth_secret: Secret,
+) -> None:
+    result = cli_runner(["neuro-extras", "init-aliases"])
+    assert result.returncode == 0, result
+
+    dockerfile_path = Path("nested/custom.Dockerfile")
+    dockerfile_path.parent.mkdir(parents=True)
+
+    random_file_to_disable_layer_caching = gen_random_file(dockerfile_path.parent)
+    with open(dockerfile_path, "w") as f:
+        f.write(
+            textwrap.dedent(
+                f"""\
+                FROM ubuntu:latest
+                ADD {random_file_to_disable_layer_caching} /tmp
+                RUN echo !
+                """
+            )
+        )
+
+    tag = str(uuid.uuid4())
+
+    # WORKAROUND: Fixing 401 Not Authorized because of this problem:
+    # https://github.com/neuromation/platform-registry-api/issues/209
+    rnd = uuid.uuid4().hex[:6]
+    img_name = f"image:extras-e2e-custom-dockerfile-{rnd}"
+    img_uri_str = f"{img_name}:{tag}"
+
+    try:
+        cmd = [
+            "neuro",
+            "image-build",
+            "-e",
+            f"{dockerhub_auth_secret.name}=secret:{dockerhub_auth_secret.name}",
+            "-f",
+            str(dockerfile_path),
+            ".",
+            img_uri_str,
+        ]
+        result = cli_runner(cmd)
+        assert result.returncode == 0, result
+
+        result = repeat_until_success(["neuro", "image", "tags", img_name])
+        assert tag in result.stdout
+    finally:
+        try:
+            cli_runner(["neuro", "image", "rm", img_uri_str])
+        except Exception:
+            # Ignore exception
```

Comment:
Same comment re: logging ignored exceptions is valid across the board

Reply:
same as above, I think no need to duplicate this discussion :)